### ISSUE TITLE
Parse error when using typescript features

### ISSUE THREAD

**Issue by fabiosantoscode** [palantir/tslint#4283]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.1.6
- __Running TSLint via__: (pick one) vim-ale
#### TypeScript code being linted

```ts
type ProcessStdoutFn = (s: string) => string
```

with `tslint.json` configuration:

```json
{
  "parser": "typescript-eslint-parser",
  "extends": "tslint-config-standard"
}
```

#### Actual behavior

Fails to parse and gives me an unexpected token error for the word "ProcessStdoutFn". Other Typescript features like typing variables and function parameters aren't parsed, giving an error for the ":" character.

#### Expected behavior

Should parse this as a type assignment.

**Comment by JoshuaKGoldberg** [comment]
Hi, thanks for reporting this bug! It looks like it's an issue with https://github.com/eslint/typescript-eslint-parser, not TSLint. Could you please file this issue there, and include the exact unexpected token error you're seeing?

**Comment by fabiosantoscode** [comment]
But, I don't even have that module installed. Should I have it?

**Comment by JoshuaKGoldberg** [comment]
> don't even have that module installed

I'm confused. What are the contents of your `package.json` and what is the exact error you're getting? If `typescript-eslint-parser` is not listed in your `package.json`, does a folder exist at `./node_modules/typescript-eslint-parser`?

The `"parser"` field is not one recognized by `tslint.json`. Did you copy and paste an `.eslintrc.json` or similar?

**Comment by fabiosantoscode** [comment]
I've removed the "parser" field. Thanks. I must have copied it from somewhere.

I checked the node_modules folder, it's not there...

**Comment by fabiosantoscode** [comment]
I've installed the module you mention, and there's no effect. The error remains.

**Comment by JoshuaKGoldberg** [comment]
@fabiosantoscode there's really nothing we can do here without more information from you. A few questions:
* **What is the exact error you're getting**? Please copy and paste it here.
* How exactly are you running TSLint (how have you configured vim-ale)?
* Does this error happen if you run something on the command line, and if so, what is the command you're using?

**Comment by fabiosantoscode** [comment]
Maybe the problem here is vim-ale. The bug also occurs on the CLI. 

The exact output is as follows:

> tslint --project src/*

src/index.ts(1,1): error TS1005: '{' expected.

**Comment by JoshuaKGoldberg** [comment]
> src/index.ts(1,1): error TS1005: '{' expected.

Ok, that's progress! Can you copy & paste the contents of your `src/index.ts` here? If not for legal/permission/secrecy/etc. reasons, how about the first few lines?

**Comment by JoshuaKGoldberg** [comment]
Oh, wait, hang on - that's not the correct way to run `--project`. `--project` takes in a path to a `tsconfig.json` file, which is expected to be valid JSON. You likely meant `tslint --project tsconfig.json`. Unrelated.

@fabiosantoscode I'm going to close this issue as external because it seems like it's a problem with your vim-ale setup. Feel free to post here if you think that's not the case.

**Comment by fabiosantoscode** [comment]
Thanks for your help :)
